### ISSUE TITLE
Add stdout of the test logger to xUnit output

### ISSUE THREAD

**Issue by ktbarrett** [cocotb/cocotb#2719]
This was removed accidentally a while ago and now there is no way to determine what argument values TestFactory were passed, or the stacktrace that caused the failure. xUnit purposefully supports capturing stdout and I think the best way of accomplishing that is by capturing the log messages emitted by the `cocotb.*` namespace logger, or maybe just the logger specifically for the test object.

**Comment by tymonx** [comment]
This will require to add a capture mechanism for logger similar to what pytest is doing. Registering a log handler that will tee content of logged messages to separate string buffer `StringIO` and dump them to `system-out` and `system-err` XML elements of test case.

Currently out of scope of this PR https://github.com/cocotb/cocotb/pull/5219 but I will try to keep this in mind and prepare a separate PR 🤔

**Comment by ktbarrett** [comment]
Yeah lets not do general purpose stdout capturing. HUGE and mostly not valuable overhead associated with every log message. Also deciding what logger to capture is difficult and requires adding yet another knob.

`TestFactory` generates a docstring that contains the values passed. Or did at one point. Is a test description a property that could be added that we could fill in with the test's docstring?

**Comment by ktbarrett** [comment]
Let's close this and track #2548.
